File: src/main/java/com/example/stopgap/generators/web/GeneratorEndpoint.java
package com.example.stopgap.generators.web;

import com.example.stopgap.Endpoint;
import com.example.stopgap.instanceregistry.InstanceRegistry;
import io.helidon.webserver.http.HttpRules;
import io.helidon.webserver.http.HttpService;
import io.helidon.webserver.http.ServerRequest;
import io.helidon.webserver.http.ServerResponse;

import java.security.SecureRandom;

public final class GeneratorEndpoint implements Endpoint {

    private final SecureRandom random = new SecureRandom();

    @Override
    public HttpService routes(final InstanceRegistry registry) {
        return (final HttpRules rules) -> rules
            .get("/number", this::randomNumber);
    }

    private void randomNumber(
        final ServerRequest req,
        final ServerResponse res
    ) {
        res.send(random.nextLong());
    }

}

File: app/src/main/kotlin/dev/sku20/stopgap/app/generator/web/GeneratorEndpoint.kt
package dev.sku20.stopgap.app.generator.web

import dev.sku20.stopgap.app.generator.StaticGenerator
import dev.sku20.stopgap.helidon.endpoint.Endpoint
import dev.sku20.stopgap.helidon.endpoint.Get
import io.helidon.webserver.http.ServerRequest
import io.helidon.webserver.http.ServerResponse
import java.security.SecureRandom

@Endpoint("/generate")
class GeneratorEndpoint(
    private val staticGenerator: StaticGenerator
) {

    private val random = SecureRandom()

    @Get("/number")
    fun randomNumber(
        req: ServerRequest,
        res: ServerResponse
    ) {
        res.send(random.nextLong().toString())
    }

    @Get("/static")
    fun staticGen(
        req: ServerRequest,
        res: ServerResponse
    ) {
        res.send(staticGenerator.value)
    }
}
